### PR TITLE
Model status will never update if the runtime pod is unschedulable

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1548  

## Description
The status displays an error with  "insufficient resources" messages instead of just loading infinitely

Screenshot from 2023-11-06 12-05-24

Even displays in the global model serving

Screenshot from 2023-11-06 12-09-24

## How Has This Been Tested?

1. Deploy a model via model serving using more resources than currently available
2. Wait for its status in the list of deployed models to become either pass or fail

## Test Impact
Added

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 1]
```diff
@@ -27,3 +35,25 @@ export const getInferenceServiceProjectDisplayName = (
   const project = projects.find(({ metadata: { name } }) => name === is.metadata.namespace);
   return project ? getProjectDisplayName(project) : 'Unknown';
 };
+
+export const checkModelStatus = (model: PodKind) => {
```
Comment (dpanshug) [inline frontend/src/pages/modelServing/screens/global/utils.ts:39]: Add return type

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 2]
```diff
@@ -27,3 +35,25 @@ export const getInferenceServiceProjectDisplayName = (
...
+  const modelStatus = model.status.conditions.some((model) => model?.reason === 'Unschedulable');
+  return {
+    FailedToSchedule: model.status.phase === 'Pending' && modelStatus,
+  };
+};
+
+export const useModelStatus = (namespace: string, name: string, isKserve: boolean) => {
```
Comment (dpanshug) [inline frontend/src/pages/modelServing/screens/global/utils.ts:46]: Move this hook to separate file and add return type to function

File: frontend/src/pages/modelServing/__tests__/modelPod.spec.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { checkModelStatus } from '~/pages/modelServing/screens/global/utils';
+import { mockUnscheduledPodK8sResource, mockPodK8sResource } from '~/__mocks__/mockPodK8sResource';
+
+const mockModelStatus = (status: boolean) => ({
+  FailedToSchedule: status,
```
Comment (dpanshug) [inline frontend/src/pages/modelServing/__tests__/modelPod.spec.ts:5]:
```suggestion
  failedToSchedule: status,
```
make this in camel case

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 4]
```diff
@@ -1,6 +1,14 @@
+import * as React from 'react';
 import { InferenceServiceKind, ProjectKind, SecretKind } from '~/k8sTypes';
 import { getDisplayNameFromK8sResource, getProjectDisplayName } from '~/pages/projects/utils';
 import { InferenceServiceModelState } from '~/pages/modelServing/screens/types';
+import { getPodsForKserve, getPodsForModelMesh } from '~/api';
+import useFetchState from '~/utilities/useFetchState';
+import { PodKind } from '~/k8sTypes';
+
+type ModelStatus = {
+  FailedToSchedule: boolean;
```
Comment (dpanshug) [inline frontend/src/pages/modelServing/screens/global/utils.ts:10]: also here, make this camel case

**Review by dpanshug — CHANGES_REQUESTED** [review]

File: frontend/src/__mocks__/mockPodK8sResource.ts [hunk 5]
```diff
@@ -428,3 +428,411 @@ export const mockPodK8sResource = ({
     qosClass: 'Burstable',
   },
 });
+
+export const mockUnscheduledPodK8sResource = ({
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply frontend/src/__mocks__/mockPodK8sResource.ts:432]: Just use the other one and parametrize the attributes there.

**Review by lucferbux — CHANGES_REQUESTED** [review]
Logic works really great, I've tried with both kserve and modelmesh and it's working as expected, just take a look at the mocking data, we can improve that part a little bit, once that's done we can merge it!

**Review by lucferbux — APPROVED** [review]
/lgtm

Great work!

**Pull request merged**